src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_ARRAY_SIZE = 10;
+    private static final double RESIZE_FACTOR = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[INITIAL_ARRAY_SIZE];
+        this.size = 0;
```

Comment:
if you have read about default value of int variables, you should know there's no need to initialize them with zero, it's done by Java already
```suggestion
```